pkg/apis/kudo/v1beta1/instance_types.go

```diff
@@ -78,22 +78,55 @@ type AggregatedStatus struct {
 type PlanStatus struct {
 	Name            string                `json:"name,omitempty"`
 	Status          ExecutionStatus       `json:"status,omitempty"`
+	Message         string                `json:"message,omitempty"`
 	LastFinishedRun metav1.Time           `json:"lastFinishedRun,omitempty"`
 	Phases          []PhaseStatus         `json:"phases,omitempty"`
 	UID             apimachinerytypes.UID `json:"uid,omitempty"`
 }
 
 // PhaseStatus is representing status of a phase
 type PhaseStatus struct {
-	Name   string          `json:"name,omitempty"`
-	Status ExecutionStatus `json:"status,omitempty"`
-	Steps  []StepStatus    `json:"steps,omitempty"`
+	Name    string          `json:"name,omitempty"`
+	Status  ExecutionStatus `json:"status,omitempty"`
+	Message string          `json:"message,omitempty"`
+	Steps   []StepStatus    `json:"steps,omitempty"`
 }
 
 // StepStatus is representing status of a step
 type StepStatus struct {
-	Name   string          `json:"name,omitempty"`
-	Status ExecutionStatus `json:"status,omitempty"`
+	Name    string          `json:"name,omitempty"`
+	Message string          `json:"message,omitempty"`
+	Status  ExecutionStatus `json:"status,omitempty"`
+}
+
+func (s *StepStatus) SetStatus(status ExecutionStatus) {
```

Comment:
naming nit: I'm not a big fan of `Set`ters and `Get`ter in method names in Go. Or maybe it's **just** `Set` and `SetWithMessage` e.g. `phaseStatus.Set(ERROR, "something bad happened")?